flink-runtime/src/main/java/org/apache/flink/runtime/state/metrics/AbstractLatencyTrackingStateMetric.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state.metrics;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.metrics.Histogram;
+import org.apache.flink.metrics.HistogramStatistics;
+import org.apache.flink.metrics.MetricGroup;
+
+import com.codahale.metrics.SlidingTimeWindowReservoir;
+import com.codahale.metrics.Snapshot;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+/** Abstract class of latency tracking state metric which counts and histogram the state metric. */
+class AbstractLatencyTrackingStateMetric implements AutoCloseable {
+    protected static final String STATE_CLEAR_LATENCY = "stateClearLatency";
+    private final MetricGroup metricGroup;
+    private final int sampleInterval;
+    private final Map<String, Histogram> histogramMetrics;
+    private final Map<String, Counter> countersPerMetric;
+    private final Supplier<com.codahale.metrics.Histogram> histogramSupplier;
```

Comment:
Why not `org.apache.flink.metrics.Histogram`?

Reply:
This is because actually we need the `com.codahale.metrics.Histogram` instance to be wrapped for the interface of `org.apache.flink.metrics.Histogram`.

Reply:
I would not need to depend on `com.codahale.metrics.Histogram` anymore.